From b9634016466b1213e078a9e23110100aa91bc474 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Wed, 13 Mar 2024 17:40:37 +0200
Subject: [PATCH] [SW-165981] emb cmn: cleaned up baremetal makefiles

Removed the redundancy PHY_FW_SIZE flag, corrected HL_DEV_ARC and
HL_DEV_ARC_COMMON path and including HL_DEV_ARC/Makfile only after
initialized all compiler & linker flags and source code files to
avoid missing the files or flags.

Change-Id: I93737d49df330aafed2c8c160de99b52b96d277e
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 9d039b4..4d51ca3 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -7,9 +7,6 @@
 HL_DEV_ARCH_COMMON := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/common
 SPECS_DEV := $(shell echo $(HL_DEV) | cut -d'_' -f1)
 
-include $(HL_DEV_ARCH_COMMON)/Makefile
-include $(HL_DEV_ARCH)/Makefile
-
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
@@ -38,9 +35,8 @@
 
 EROM_PCIE_SRC := erom_pcie.o
 EROM_RECOVERY_SRC := erom_recovery.o
-LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports.elf
 
-EROM_PCIE_DEPS += \
+EROM_PCIE_DEPS := \
 	$(PREBOOT_ROOT_DIR)/lib/common/pll.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/pci.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/uart.o \
@@ -55,7 +51,7 @@
 	$(HL_DEV_ARCH)/../common/security.o \
 	erom_secondary.o
 
-EROM_RECOVERY_DEPS += \
+EROM_RECOVERY_DEPS := \
 	$(PREBOOT_ROOT_DIR)/lib/common/boot_desc.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/cmd_module.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/module_i2c_slave.o \
@@ -93,6 +89,8 @@
 	$(HL_DEV_ARCH_COMMON)/reset.o \
 	$(HL_DEV_ARCH)/$(HL_TARGET).o
 
+include $(HL_DEV_ARCH)/Makefile
+
 EROM_RECOVERY_EXTRA_LIBS := "-lfdt"
 
 .PHONY: all
diff --git a/app/pcie_pif/Makefile b/app/pcie_pif/Makefile
index 4289855..a3f4650 100644
--- a/app/pcie_pif/Makefile
+++ b/app/pcie_pif/Makefile
@@ -2,11 +2,9 @@
 OBJCOPY = $(CROSS_COMPILE)objcopy
 BASE_DEV_NAME := $(shell echo $(HL_DEV) | cut -d'_' -f1)
 HL_DEV_SUBTYPE := $(shell echo $(HL_DEV) | cut -d'_' -s -f2,3)
-HL_DEV_ARCH := arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
-HL_DEV_ARCH_COMMON := arch/$(BASE_DEV_NAME)/common
 PREBOOT_ROOT_DIR := ../..
-
-include $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/Makefile
+HL_DEV_ARCH := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
+HL_DEV_ARCH_COMMON := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/common
 
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
@@ -14,12 +12,9 @@
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c
-CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/
+CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(HL_DEV_ARCH)/
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
-
-include $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/pcie_phy_fw.mk
-CFLAGS += -DPHY_FW_SIZE=$(PHY_FW_SIZE)
 
 SFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
@@ -30,11 +25,13 @@
 
 PIF_ARC_SRCS_C := $(shell ls *.c)
 PIF_ARC_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/lib/common/*.c)
-PIF_ARC_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/common.c)
-PIF_ARC_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/common.c)
-PIF_ARC_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/$(HL_TARGET).c)
+PIF_ARC_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/common.c)
+PIF_ARC_SRCS_C += $(shell ls $(HL_DEV_ARCH)/common.c)
+PIF_ARC_SRCS_C += $(shell ls $(HL_DEV_ARCH)/$(HL_TARGET).c)
 PIF_ARC_APP_S := $(shell ls crt0_pcie_pif.S)
 
+include $(HL_DEV_ARCH)/Makefile
+
 PIF_ARC_OBJECTS := $(PIF_ARC_SRCS_C:.c=.o)
 PIF_ARC_OBJECTS += $(PIF_ARC_APP_S:.S=.o)
 
diff --git a/app/pldm_bloader/Makefile b/app/pldm_bloader/Makefile
index 0ec381a..ff94d37 100644
--- a/app/pldm_bloader/Makefile
+++ b/app/pldm_bloader/Makefile
@@ -6,8 +6,6 @@
 HL_DEV_ARCH := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
 HL_DEV_ARCH_COMMON := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/common
 
-include $(HL_DEV_ARCH_COMMON)/Makefile
-
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(BASE_DEV_NAME)
@@ -18,19 +16,11 @@
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
 
-include $(HL_DEV_ARCH_COMMON)/pcie_phy_fw.mk
-CFLAGS += -DPHY_FW_SIZE=$(PHY_FW_SIZE)
-
 SFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
 PLDM_BLOADER_LD_SCRIPT := $(shell ls $(HL_DEV_ARCH)/pldm_bloader.lds.S)
 LDFLAGS := -g -nostdlib -static
 LDFLAGS += -Wl,--gc-sections
-ARBOOT_EXPORTS_FILE := $(HL_DEV_ARCH)/exports.elf
-ifeq ($(shell test -e $(ARBOOT_EXPORTS_FILE) && echo -n yes),yes)
-LDFLAGS += -Wl,--just-symbols $(ARBOOT_EXPORTS_FILE)
-endif
-
 LDLIBS := -lgcc
 
 PLDM_BLOADER_SRCS_C := $(shell ls *.c)
@@ -40,6 +30,8 @@
 PLDM_BLOADER_SRCS_C += $(shell ls $(HL_DEV_ARCH)/$(HL_TARGET).c)
 PLDM_BLOADER_APP_S := $(shell ls crt0_rom.S)
 
+include $(HL_DEV_ARCH)/Makefile
+
 PLDM_BLOADER_OBJECTS := $(PLDM_BLOADER_SRCS_C:.c=.o)
 PLDM_BLOADER_OBJECTS += $(PLDM_BLOADER_APP_S:.S=.o)
 
diff --git a/arch/fs1/arc/Makefile b/arch/fs1/arc/Makefile
index 36ba315..fe1ff63 100644
--- a/arch/fs1/arc/Makefile
+++ b/arch/fs1/arc/Makefile
@@ -1,3 +1,5 @@
+include $(HL_DEV_ARCH_COMMON)/Makefile
+
 include $(HL_DEV_ARCH)/pcie_phy_fw.mk
 
 PPBOOT_MAX_SIZE := 16384
@@ -7,6 +9,12 @@
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/fs/common
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/fs/$(DEV_PREFIX)
 
+# TODO: Remove when bootrom exports is available
+ARBOOT_EXPORTS_FILE := $(HL_DEV_ARCH)/exports.elf
+ifeq ($(shell test -e $(ARBOOT_EXPORTS_FILE) && echo -n yes),yes)
+LDFLAGS += -Wl,--just-symbols $(ARBOOT_EXPORTS_FILE)
+endif
+
 PPBOOT_SRCS_C := $(filter-out lib/common/module_i2c_slave.c,$(PPBOOT_SRCS_C))
 PPBOOT_SRCS_C := $(filter-out lib/common/pif_ts.c,$(PPBOOT_SRCS_C))
 
diff --git a/arch/fs1/common/Makefile b/arch/fs1/common/Makefile
index 7b6fbcc..208d225 100644
--- a/arch/fs1/common/Makefile
+++ b/arch/fs1/common/Makefile
@@ -1,3 +1 @@
-include $(HL_DEV_ARCH)/Makefile
-
 # this Makefile is for backwards compatibility, please use ARC Makefile
diff --git a/arch/gaudi3/arc/Makefile b/arch/gaudi3/arc/Makefile
index c3e02e0..ba3f472 100644
--- a/arch/gaudi3/arc/Makefile
+++ b/arch/gaudi3/arc/Makefile
@@ -1,7 +1,11 @@
 include $(HL_DEV_ARCH_COMMON)/Makefile
-include $(HL_DEV_ARCH_COMMON)/pcie_phy_fw.mk
 
 PPBOOT_MAX_SIZE := 18432
+
+include $(HL_DEV_ARCH_COMMON)/pcie_phy_fw.mk
+CFLAGS += -DPHY_FW_SIZE=$(PHY_FW_SIZE)
+
+CFLAGS += -I $(PREBOOT_ROOT_DIR)/lib/preboot
 
 ifeq ($(HL_TARGET),onix)
 	LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports_onix.elf
@@ -9,18 +13,15 @@
 	LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports.elf
 endif
 
-CFLAGS += -I $(PREBOOT_ROOT_DIR)/lib/preboot
-CFLAGS += -DPHY_FW_SIZE=$(PHY_FW_SIZE)
-
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/mbist.c)
 
-EROM_PCIE_DEPS := \
+EROM_PCIE_DEPS += \
 	$(PREBOOT_ROOT_DIR)/lib/common/boot_desc.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/efuse.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/string.o \
 	$(HL_DEV_ARCH_COMMON)/pci.o
 
-EROM_RECOVERY_DEPS := \
+EROM_RECOVERY_DEPS += \
 	$(HL_DEV_ARCH_COMMON)/pci.o \
 	$(HL_DEV_ARCH_COMMON)/fit.o
 
